packages/stack/contracts-manager/src/index.js

```diff
@@ -353,15 +356,16 @@ export default class ContractsManager {
 
         for (className in self.contracts) {
           contract = self.contracts[className];
-          contract.deploy = (contract.deploy === undefined) || contract.deploy;
           if (self.deployOnlyOnConfig && !contractsConfig.contracts[className]) {
             contract.deploy = false;
           }
 
-          if (!contractsConfig.contracts[className] && contractsConfig.strategy === constants.deploymentStrategy.explicit) {
+          if (contract.deploy !== true && !contractsConfig.contracts[className] && contractsConfig.strategy === constants.deploymentStrategy.explicit) {
             contract.deploy = false;
           }
 
+          contract.deploy = (contract.deploy === undefined) || contract.deploy;
```

Comment:
I think the following is equivalent (if I understood the logic correctly) and maybe simpler:

```suggestion
          contract.deploy = contract.deploy ?? true;
```